packages/fx-core/src/component/resource/botService/appStudio/appStudioClient.ts

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IBotRegistration } from "./interfaces/IBotRegistration";
+
+import { AxiosInstance, AxiosResponse, default as axios } from "axios";
+import {
+  BotRegistrationNotFoundError,
+  ConfigUpdatingError,
+  MessageEndpointUpdatingError,
+  ProvisionError,
+} from "../errors";
+import { CommonStrings, ConfigNames } from "../strings";
+import { RetryHandler } from "../retryHandler";
+import { Messages } from "../messages";
+import { getAppStudioEndpoint } from "../../appManifest/constants";
+import { ResourceContextV3 } from "@microsoft/teamsfx-api";
+import { CheckThrowSomethingMissing } from "../../../error";
+import { FxBotPluginResultFactory } from "../result";
+
+export class AppStudioClient {
+  private static baseUrl = getAppStudioEndpoint();
+
+  private static newAxiosInstance(accessToken: string): AxiosInstance {
+    accessToken = CheckThrowSomethingMissing(
+      FxBotPluginResultFactory.source,
+      ConfigNames.APPSTUDIO_TOKEN,
+      accessToken
+    );
+    const instance = axios.create({
+      headers: {
+        post: {
+          Authorization: `Bearer ${accessToken}`,
+          "Client-Source": "teamstoolkit",
+        },
+        get: {
+          Authorization: `Bearer ${accessToken}`,
+          "Client-Source": "teamstoolkit",
+        },
+      },
+    });
+    instance.interceptors.request.use(function (config) {
+      config.params = { teamstoolkit: true, ...config.params };
+      return config;
+    });
+    return instance;
+  }
+
+  public static async getBotRegistration(
+    token: string,
+    botId: string
+  ): Promise<IBotRegistration | undefined> {
+    const axiosInstance = AppStudioClient.newAxiosInstance(token);
+
+    const getBotRegistrationResponse: AxiosResponse<any> | undefined = await RetryHandler.Retry(
+      async () => {
+        try {
+          return await axiosInstance.get(`${AppStudioClient.baseUrl}/api/botframework/${botId}`);
+        } catch (e) {
+          if (e.response?.status === 404) {
+            return e.response;
+          } else {
+            e.teamsfxUrlName = "<get-bot-registration>";
+            throw e;
+          }
+        }
+      },
+      true
+    );
+    if (getBotRegistrationResponse?.status === 200) {
+      return <IBotRegistration>getBotRegistrationResponse.data;
+    } else {
+      return undefined;
+    }
+  }
+
+  public static async createBotRegistration(
+    token: string,
+    registration: IBotRegistration,
+    context?: ResourceContextV3
+  ): Promise<void> {
+    const axiosInstance = AppStudioClient.newAxiosInstance(token);
+
+    let response = undefined;
+    try {
+      if (registration.botId) {
+        const botReg = await AppStudioClient.getBotRegistration(token, registration.botId);
+        if (botReg) {
```

Comment:
If botId is not found (404), the `AppStudioClient.getBotRegistration` function will return error.response from L61. Seem the `if(botReg)` in L88 will return true.

Reply:
Good catch! Will update it soon.

Reply:
After a second check, `getBotRegistration` will return undefined if 404 was met.